src/components/App/App.js

```diff
@@ -1,38 +1,54 @@
 import React, {Component} from 'react'
 import Movies from '../Movies/Movies.js'
 import MovieDetails from '../MovieDetails/MovieDetails'
-import movieData from '../../movieData'
+// import movieData from '../../movieData'
 import './App.css'
+import { getAllMovies, getSingleMovie } from '../../apiCalls'
 
 class App extends Component {
   constructor() {
     super()
     this.state = {
-      movieData: movieData.movies,
+      movieData: [],
       displayMovieDetails: false,
-      id: ''
+      error: '',
+      singleMovie: {},
     }
   }
 
-  toggleMovieDetails = id => {
+  componentDidMount = () => {
+    getAllMovies()
+      .then(data => {
+        this.setState({ movieData: data.movies })
+      })
+      .catch(error => this.setState({ error: 'Something went wrong, try again later!' }))
+  }
 
-    this.setState({displayMovieDetails: !this.state.displayMovieDetails, id: id})
+  toggleMovieDetails = id => {
```

Comment:
I wonder if this function is necessary. Seems like it simply calls `fetchSingleMovie` - perhaps whatever invokes this function could simply invoke `fetchSingleMovie` instead.